portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -66,51 +70,57 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     public static class UserData {
         long id;
         String name;
-        String email;
         String comment;
         Date date;
-
-        UserData(long id, String name, String email, String comment, Date date) {
+    
+        UserData(long id, String name, String comment, Date date) {
             this.id = id;
             this.name = name;
-            this.email = email;
             this.comment = comment;
             this.date = date;
         }
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Date date = new Date();
-        String name = getParameter(request, "name", "");
-        String email = getParameter(request, "email", "");
-        String comment = getParameter(request, "comment", "");
-
-        if (name.trim().isEmpty()) {
-            throw new IllegalArgumentException("Name not entered.");
+        // User must be logged in before posting comments.
+
+        if (userService.isUserLoggedIn()) {
+            Date date = new Date();
+            String name = getParameter(request, "name", "");
+            String comment = getParameter(request, "comment", "");
+            String userEmail = userService.getCurrentUser().getEmail();
+
+            if (name.trim().isEmpty()) {
+                throw new IllegalArgumentException("Name not entered.");
+            }
+
+            if (comment.trim().isEmpty()) {
+                throw new IllegalArgumentException("Comment field empty.");
+            }
+
+            Entity taskEntity = new Entity("UserData");
+            taskEntity.setProperty("name", name);
+            taskEntity.setProperty("email", userEmail);
+            taskEntity.setProperty("comment", comment);
+            taskEntity.setProperty("date", date);
+            datastore.put(taskEntity);
+
+            // Redirect back to the HTML page.
+            response.sendRedirect("/index.html");
         }
```

Comment:
"} else {" - should all be on the same line

Reply:
done